File: 2015/03-20/Pat/FizzBuzz.swift
#!/usr/bin/swift
/**
 * FizzBuzz with modulo operation
 *
 * @author      Pat Hawks
 * Created on:  Mar 19, 2015
 * Source File: FizzBuzz.swift
 *
 * To run:
 *     swift FizzBuzz.swift
 * Or to compile:
 *     swiftc -sdk $(xcrun --show-sdk-path --sdk macosx) FizzBuzz.swift
 */

var fizz : Bool
var buzz : Bool

for i in 1...100 {
    fizz = (i % 3 == 0)
    buzz = (i % 5 == 0)

    switch true {
    case fizz && buzz:
        print("FizzBuzz")     // Look ma, no breaks!
    case fizz:
        print("Fizz")
    case buzz:
        print("Buzz")
    default:
        print(i)
    }
}

File: 2015/03-20/Pat/FizzBuzz.c
/**
 * FizzBuzz with no modulo operation
 *
 * @author      Pat Hawks
 * Created on:  Mar 19, 2015
 * Source File: FizzBuzz.c
 *
 * To compile:
 *    gcc FizzBuzz.c -o FizzBuzz
 */

#include <stdio.h>

int main( void ) {
    const char* FIZZ = "Fizz";
    const char* BUZZ = "Buzz";

    for (unsigned int i=1, j=1; i<101; ++i, ++j) {
        switch (j) {
            case 15:
                printf("%s%s\n", FIZZ, BUZZ);
                j=1, ++i;
                break;
            case  9:
                printf("%s\n%s\n", FIZZ, BUZZ);
                ++i, ++j;
                continue;
            case  5:
                printf("%s\n", BUZZ);
                ++i, ++j;
            case  3:
            case 12:
                printf("%s\n", FIZZ);
                ++i, ++j;
        }
        printf("%d\n", i);
    }
}
